Source/Core/Duality/Utility/Coroutines/WaitUntil.cs

```diff
@@ -0,0 +1,94 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Duality.Utility.Coroutines
+{
+	/// <summary>
+	/// Struct in charge of managing the timings of a Coroutine's execution
+	/// </summary>
+	public struct WaitUntil
+	{
+		private enum WaitType
```

Comment:
This can be a byte instead of a int to save some memory (though I am afraid the compiler will likely pad this struct to be 8 bytes big anyway).

Reply:
its still a int?

Reply:
Yeah, I tried to change it to byte but in the end it still takes 8 bytes in memory